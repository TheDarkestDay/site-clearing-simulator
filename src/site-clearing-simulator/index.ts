export { default as siteClearingSimulatorReducer } from './store/store-slice';
export { SiteMap } from './site-map/SiteMap';