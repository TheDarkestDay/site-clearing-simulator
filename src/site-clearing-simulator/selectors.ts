import { GlobalState } from '../store';

export const getSiteMap = (state: GlobalState) => state.siteClearingSimulator.map;

export const isSimulationStarted = (state: GlobalState) => state.siteClearingSimulator.isStarted;