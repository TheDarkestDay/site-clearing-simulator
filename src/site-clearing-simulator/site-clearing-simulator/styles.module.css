.simulatorWrapper {
  padding: 16px;
}