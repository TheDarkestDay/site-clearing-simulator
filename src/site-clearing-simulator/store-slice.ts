import { createSlice } from "@reduxjs/toolkit";

export type SiteCellType = "o" | "t" | "r" | "T" | "B" | "C";

export enum BulldozerDirection {
  Up = 1,

  Right = 2,

  Down = 3,

  Left = 4,
}

export type SiteClearingSimulatorState = {
  map: SiteCellType[][];
  isStarted: boolean;
  isStopped: boolean;
  isPreservedTreeRemoved: boolean;
  bulldozerPosition: {
    x: number;
    y: number;
  };
  bulldozerDirection: BulldozerDirection;
  fuelUsed: number;
};

// ["o", "o", "t", "o", "o", "o", "o", "o", "o", "o"],
//     ["o", "o", "o", "o", "o", "o", "o", "T", "o", "o"],
//     ["r", "r", "r", "o", "o", "o", "t", "T", "o", "o"],
//     ["r", "r", "r", "r", "o", "o", "o", "o", "o", "o"],
//     ["r", "r", "r", "r", "r", "t", "o", "o", "o", "o"]

const initialState: SiteClearingSimulatorState = {
  map: [],
  isStarted: false,
  isStopped: false,
  bulldozerDirection: BulldozerDirection.Right,
  bulldozerPosition: {
    x: -1,
    y: -1,
  },
  isPreservedTreeRemoved: false,
  fuelUsed: 0,
};

const siteClearingSimulatorSlice = createSlice({
  name: "siteClearingSimulator",
  initialState,
  reducers: {
    startSimulation(state, action) {},
    stopSimulation() {},
    moveForward(state) {
      return state;
    },
    rotateLeft(state) {
      return state;
    },
    rotateRight(state) {
      return state;
    },
    setMap(state, action) {
      return state;
    },
  },
});

export const {
  startSimulation,
  stopSimulation,
  moveForward,
  rotateLeft,
  rotateRight,
  setMap,
} = siteClearingSimulatorSlice.actions;

export default siteClearingSimulatorSlice.reducer;
