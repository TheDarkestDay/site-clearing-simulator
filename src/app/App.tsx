import React from 'react';
import { SiteClearingSimulator } from '../site-clearing-simulator/site-clearing-simulator';
import styles from './styles.module.css';

function App() {
  return (
    <div className={styles.app}>
      <SiteClearingSimulator />
    </div>
  );
}

export default App;
