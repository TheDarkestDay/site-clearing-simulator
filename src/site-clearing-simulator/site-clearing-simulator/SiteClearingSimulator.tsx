import { useSelector } from 'react-redux';
import { isSimulationStarted } from '../selectors';
import { SimulationFileInput } from '../simulation-file-input/SimulationFileInput';
import { SimulatorWindow } from '../simulator-window/SimulatorWindow';

export const SiteClearingSimulator = () => {
  const isStarted = useSelector(isSimulationStarted);

  return (
    <main>
      {
        isStarted
          ? <SimulatorWindow />
          : <SimulationFileInput />
      }
    </main>
  );
};