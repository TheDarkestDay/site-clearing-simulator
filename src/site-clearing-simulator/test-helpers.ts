import { SiteClearingSimulatorState, BulldozerDirection } from './store/store-slice';

export const getTestState = (
  overrides: Partial<SiteClearingSimulatorState> = {}
): SiteClearingSimulatorState => {
  return {
    map: [],
    isStarted: false,
    isStopped: false,
    bulldozerDirection: BulldozerDirection.Right,
    bulldozerPosition: {
      x: 0,
      y: -1,
    },
    isPreservedTreeRemoved: false,
    fuelUsed: 0,
    eventsLog: [],
    ...overrides,
  };
};