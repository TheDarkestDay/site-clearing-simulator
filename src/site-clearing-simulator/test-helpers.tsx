import { createStore } from '@reduxjs/toolkit';
import { create } from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { GlobalState, rootReducer } from '../store';
import { SiteClearingSimulatorState, BulldozerDirection, initialState } from './store/store-slice';

export const getTestState = (
  overrides: Partial<SiteClearingSimulatorState> = {}
): SiteClearingSimulatorState => {
  return {
    map: [],
    isStarted: false,
    isStopped: false,
    bulldozerDirection: BulldozerDirection.Right,
    bulldozerPosition: {
      x: 0,
      y: -1,
    },
    isPreservedTreeRemoved: false,
    fuelUsed: 0,
    eventsLog: [],
    ...overrides,
  };
};

export const renderWithStore = (component: React.ReactNode, state: Partial<GlobalState['siteClearingSimulator']> = {}) => {
  const testStore = createStore(rootReducer, {
    siteClearingSimulator: {
      ...initialState,
      ...state,
    }
  });
  const consumer = (
    <Provider store={testStore}>
      {component}
    </Provider>
  );

  return create(consumer);
}