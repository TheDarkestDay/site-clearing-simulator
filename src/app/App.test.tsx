import React from 'react';
import { renderWithStore } from '../site-clearing-simulator/test-helpers';
import App from './App';

describe('App component', () => {
  it('should match snapshot', () => {
    const wrapper = renderWithStore(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});