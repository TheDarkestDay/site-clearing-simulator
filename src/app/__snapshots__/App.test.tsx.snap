// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<div
  className="app"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h1
        className="MuiTypography-root MuiTypography-h6"
      >
        Site Clearing Simulator
      </h1>
    </div>
  </header>
  <main
    className="simulatorWrapper"
  >
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root"
      >
        <form>
          <label
            className="MuiTypography-root fileFieldLabel MuiTypography-body1 MuiTypography-gutterBottom"
            htmlFor="simulationFile"
          >
            Please, provide a file to start simulation:
          </label>
          <input
            accept="text/plain"
            id="simulationFile"
            onChange={[Function]}
            type="file"
          />
        </form>
      </div>
    </div>
  </main>
</div>
`;
